todolist/parser_test.go

```diff
@@ -74,6 +74,32 @@ func TestParseContexts(t *testing.T) {
 	}
 }
 
+func TestParseNotes(t *testing.T) {
+	parser := &Parser{}
+	todo := parser.ParseNewTodo("add search engine survey")
+
+	if parser.ParseNotes(todo, "an 1 www.google.com") != "add" {
+		t.Error("Expected Notes to be added")
+	}
+	if todo.Notes[0] != "www.google.com" {
+		t.Error("Expected note 1 to be 'www.google.com' but got", todo.Notes[0])
+	}
+
+	if parser.ParseNotes(todo, "en 1 0 www.duckduckgo.com") != "edit" {
+		t.Error("Expected Notes to be editted")
+	}
+	if todo.Notes[0] != "www.duckduckgo.com" {
+		t.Error("Expected note 1 to be 'www.duckduckgo.com' but got", todo.Notes[0])
+	}
+
+	if parser.ParseNotes(todo, "dn 1 0") != "delete" {
+		t.Error("Expected Notes to be deleted")
+	}
+	if len(todo.Notes) != 0 {
+		t.Error("Expected no note")
+	}
```

Comment:
The above tests are for failure cases only.  I'd love to see tests for the happy path for each note subcommand.

Reply:
Would you provide more explanation?  I know what happy path is but I don't see why `TestParseNotes`is not.  How about other test functions such as `TestParseProjects`?

Also, in the refine patch, I add three more test functions to focus the testing on each `Parse*` functions.

Reply:
Now three more `TestParse*Note` functions are available.  Is that what you mean?